YeRoBattle.Engine/BattleCalculator.cs

```diff
@@ -31,6 +32,62 @@ public void Hit(Character attacker, Character defender)
             {
                 defender.IsDead = true;
             }
+        }
+        public void Healing(Character character, Character target)
+        {
+            var heal = character.HealPower;
+
+            if (heal > 0) // heal character 
+            {
+                _logger.WriteLine($@"character {character.Name} has healed {character.Name} by {heal}");
+                character.Health = character.Health + heal;
+            }
+            if (character.CurrentHealth == character.Health)
+            {
+                _logger.WriteLine($@"character {character.Name} has healed {character.Name} by {heal}");
+                character.HealPower = 0;
+            }
+            else
+            {
+                _logger.WriteLine($@"character {character.Name} has healed {character.Name} by {heal}");
+                character.Health = character.Health + heal;
+            }
+
+
+            if (heal > 0) //heal target
+            {
+                _logger.WriteLine($@"character {character.Name} has healed {target.Name} by {heal}");
+                target.CurrentHealth = target.CurrentHealth + heal;
+            }
+            if (target.CurrentHealth == target.Health)
+            {
+                _logger.WriteLine($@"character {character.Name} has healed {target.Name} by {heal}");
+                character.HealPower = 0;
+                target.HealPower = 0;
```

Comment:
LOL
set target.HealPower = 0 
it is very weird. What impact are you expecting here?

As I understand if you don't want to heal - just do nothing. no need to set HealPower = 0 to anybody.

Reply:
done